crates/sui-indexer/src/apis/extended_api.rs

```diff
@@ -21,6 +28,56 @@ impl<S: IndexerStore> ExtendedApi<S> {
     pub fn new(state: S) -> Self {
         Self { state }
     }
+
+    fn query_objects_internal(
+        &self,
+        query: SuiObjectResponseQuery,
+        cursor: Option<ObjectID>,
+        limit: Option<usize>,
+        at_checkpoint: Option<CheckpointId>,
+    ) -> Result<ObjectsPage, IndexerError> {
+        let limit = validate_limit(limit, QUERY_MAX_RESULT_LIMIT_OBJECTS)?;
+
+        let at_checkpoint = match at_checkpoint {
+            Some(CheckpointId::SequenceNumber(seq)) => seq,
+            Some(CheckpointId::Digest(digest)) => {
+                self.state.get_checkpoint_sequence_number(digest)?
+            }
+            None => self.state.get_latest_checkpoint_sequence_number()? as u64,
+        };
+
+        let SuiObjectResponseQuery { filter, options } = query;
+        let filter = filter.unwrap_or_else(|| SuiObjectDataFilter::MatchAll(vec![]));
```

Comment:
question: `filter.unwrap_or(SuiObjectDataFilter::MatchAll(vec![]));` instead?

Reply:
in rust we usually prefer using unwrap_or_else so the code don't need to create the object, using `unwrap_or` in this case will probably trigger a clip complain